client/blocks/upload-image/README.md

```diff
@@ -0,0 +1,42 @@
+Upload Image
+=========
+
+`UploadImage` component is used to accommodate the following flow:
+- click on a designated place to open file explorer
+- in file explorer, select an image to upload
+- image is moved to `ImageEditor` where user can edit the image before uploading
+- after clicking on "Done", `UploadImage` calls the `onImageEditorDone` handler where you can perform any operation
+ (such as actual uploading of the image).
+ 
+The initial idea to create the block comes from the `edit-gravatar` one -- it's very similar but reusable and extensible.
+
+
+#### Basic usage:
+
+```js
+import UploadImage from 'blocks/upload-image';
+
+render: function() {
+	return
+		<UploadImage onImageEditorDone={ ( imageBlob ) => console.log( URL.createObjectURL( imageBlob ) ) } />;
+}
+```
+
+To see a more complex example, have a look at `blocks/upload-image/docs/example`.
+
+#### Props
+
+- `isUploading`: (default: false) whether to display a spinner over selected-n-edited image (use only after user 
+	selects an image in `onImageEditorDone`).
+- `imageEditorProps`: (default: allowedAspectRatios set to 1X1) object of additional props to send to `ImageEditor`
+	component.
+- `texts`: (default: `{}`) object of different texts/messages to show.
```

Comment:
Yeah I see. My original idea was to use an object because there were a lot of texts to translate/pass in. But eventually, I got rid of some of them and now it's not that much.

I'll put them into separate props. :+1: